src/timeline.ts

```diff
@@ -193,87 +223,105 @@ class Timeline implements IAppView {
       svgtimeline.attr('width', widthTimelineDiv);
       d3.select('line').attr('x2', widthTimelineDiv);
 
-      //Updating scale for circle position
-      xScaleTime.range([20, widthTimelineDiv-20]);
+      // Updating scale for circle position
+      xScaleTime.range([20, widthTimelineDiv - 20]);
 
       svgtimeline.selectAll('circle')
-        .attr('cx', (d:any, i ) => {
-          if(d.time) {
-            return xScaleTime(moment(d.time).diff(moment(items[0].time),'days'));
+        .attr('cx', (d: any, i) => {
+          if (d.time) {
+            return xScaleTime(moment(d.time).diff(moment(items[0].time), 'days'));
           } else {
             return i * scaleCircles(widthTimelineDiv);
           }
         });
 
-      //Update bars
+      // Update bars
       svgtimeline.selectAll('g').remove();
 
-      if(widthTimelineDiv <= 800) {
+      if (widthTimelineDiv <= 800) {
         if (rectWidth >= 5) {
           svgtimeline.selectAll('g').remove();
         } else {
-          rectWidth = rectWidth-1;
+          rectWidth = rectWidth - 1;
           generateBars(rectWidth);
         }
       } else {
         rectWidth = 15;
         generateBars(rectWidth);
       }
-    };
+    }
 
-    //creating 2D Ratio bars
+
+
+    // helper variable for on click event
+    let open2dHistogram = null;
+
+    // creating 2D Ratio bars
     function generateBars(width) {
-     return idPairs.map((pair) => {
-      console.log('start loading pair', pair);
+      return idPairs.map((pair) => {
+        console.log('start loading pair', pair);
 
-      return Promise.all([ajax.getAPIJSON(`/taco/diff_log/${pair[0]}/${pair[1]}/1/1/2/structure,content`), pair])
-        .then((args) => {
-          const json = args[0];
-          const pair = args[1];
+        return Promise.all([ajax.getAPIJSON(`/taco/diff_log/${pair[0]}/${pair[1]}/1/1/2/structure,content`), pair])
+          .then((args) => {
+            const json = args[0];
+            const pair = args[1];
 
-          const pairPosX = pair.map((d) => parseFloat(d3.select(`#circle_${d}`).attr('cx')));
+            const pairPosX = pair.map((d) => parseFloat(d3.select(`#circle_${d}`).attr('cx')));
 
-          console.log('finished loading pair', pair, pairPosX, json);
+            console.log('finished loading pair - BARS', pair, pairPosX, json);
 
-          const w = 80;
-          const h = 30;
-          const barPadding = 0.5;
+            const w = 80;
+            const h = 30;
+            const barPadding = 0.5;
 
-          const data = [json.no_ratio, json.a_ratio, json.c_ratio, json.d_ratio];
+            const data = [json.no_ratio, json.a_ratio, json.c_ratio, json.d_ratio];
 
-          const color = d3.scale.ordinal()
-            .domain(<any>[ 0, data.length -1])
-            .range(['#D8D8D8', '#67C4A7' , '#8DA1CD', '#F08E65']);
+            const color = d3.scale.ordinal()
```

Comment:
Define constants (e.g., colors, width) in the top of the class 

```
private colorScale = d3.scale.ordinal().range(['#D8D8D8', '#67C4A7', '#8DA1CD', '#F08E65']);
```

And later just change the domain of the `colorScale` object: `this.colorScale.domain(<any>[0, data.length - 1]);`